java-source/src/main/java/com/example/flow/ExampleFlow.java

```diff
@@ -135,84 +135,67 @@ public Initiator(IOUState iou, Party otherParty) {
+            final KeyPair keyPair = getServiceHub().getLegalIdentityKey();
+            final Party notary = single(getServiceHub().getNetworkMapCache().getNotaryNodes()).getNotaryIdentity();
+            // Obtain a reference to the notary we want to use and its public key.
+            final CompositeKey notaryPubKey = notary.getOwningKey();
+
+            // Stage 5.
+            progressTracker.setCurrentStep(RECEIVING_TRANSACTION);
+            // All messages come off the wire as UntrustworthyData. You need to 'unwrap' them. This is where you
+            // validate what you have just received.
+
+            final SignedTransaction partSignedTx = receive(SignedTransaction.class, otherParty)
+                    .unwrap(tx ->
+                    {
+                        // Stage 6.
+                        progressTracker.setCurrentStep(VERIFYING_TRANSACTION);
+                        try {
+                            // Check that the signature of the other party is valid.
+                            // Our signature and the notary's signature are allowed to be omitted at this stage as
+                            // this is only a partially signed transaction.
+                            final WireTransaction wireTx = tx.verifySignatures(CryptoUtilities.getComposite(keyPair.getPublic()), notaryPubKey);
+
+                            // Run the contract's verify function.
+                            // We want to be sure that the agreed-upon IOU is valid under the rules of the contract.
+                            // To do this we need to run the contract's verify() function.
+                            wireTx.toLedgerTransaction(getServiceHub()).verify();
+                        } catch (SignatureException|TransactionResolutionException ex) {
+                            throw new RuntimeException(ex);
+                        }
+                        return tx;
+                    });
+
+            // Stage 7.
+            progressTracker.setCurrentStep(SIGNING_TRANSACTION);
+            // Sign the transaction with our key pair and add it to the transaction.
+            // We now have 'validation consensus'. We still require uniqueness consensus.
+            // Technically validation consensus for this type of agreement implicitly provides uniqueness consensus.
+            final DigitalSignature.WithKey mySig = partSignedTx.signWithECDSA(keyPair);
+            // Add our signature to the transaction.
+            final SignedTransaction signedTx = partSignedTx.plus(mySig);
+
+            // Stage 8.
+            progressTracker.setCurrentStep(FINALISING_TRANSACTION);
+            final Set<Party> participants = ImmutableSet.of(getServiceHub().getMyInfo().getLegalIdentity(), otherParty);
+            // FinalityFlow() notarises the transaction and records it in each party's vault.
+            subFlow(new FinalityFlow(signedTx, participants));
+
+            return signedTx;
```

Comment:
The Acceptor's result is never used since it's initialised by the flow framework. Best to make the return type `Void` and return null.